include/cadmium/simulation/rt_clock/mbedclock.hpp

```diff
@@ -0,0 +1,230 @@
+/**
+ * Real-time clock based on the Driver APIs from MBED-OS version 5.
+ * Copyright (C) 2023  Ezequiel Pecker Marcosig
+ * SEDLab - University of Buenos Aires
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef CADMIUM_SIMULATION_RT_CLOCK_MBED_HPP
+#define CADMIUM_SIMULATION_RT_CLOCK_MBED_HPP
+
+// #include <iostream>
+#include <chrono>
+#include <optional>
+#include <thread>
+#include "rt_clock.hpp"
+#include "../../exception.hpp"
+#include <mbed.h>
+
+static long MIN_TO_MICRO   = (1000*1000*60);
+static long SEC_TO_MICRO   = (1000*1000);
+static long MILI_TO_MICRO  = (1000);
+
+namespace cadmium {
+    /**
+     * Real-time clock based on the Driver APIs provided by MBED-OS version 5: https://os.mbed.com/docs/mbed-os/v5.15/apis/drivers.html.
+     * @tparam T Internal clock type. By default, it uses a double.
+     */
+    template <typename T = double>
+    class MBEDClock: RealTimeClock {
+     private:
+	// executionTimer is used in the waitUntil() to measure the elapsed time between cnsecutive calls from the Root Coordinator and the time elapsed between an event and an external interruption
```

Comment:
Please, review the docs and use Doxygen formatting. The docs are automatically generated by GitHub based on this format.